src/lualdap.c

```diff
@@ -981,9 +981,15 @@ static int lualdap_open_simple (lua_State *L) {
 	if (strstr(host, "://") != NULL) {
 		err = ldap_initialize(&conn->ld, host);
 	} else {
-		const char *host_with_schema = lua_pushfstring(L, "ldap://%s", host);
-		err = ldap_initialize(&conn->ld, host_with_schema);
-		lua_pop(L, 1);
+		lua_getglobal (L, "string");
```

Comment:
Should the string library be loaded from C instead of relying on a global being present?

```suggestion
		luaL_requiref (LUA_STRLIBNAME, luaopen_string);
```

Reply:
If an user remove the standard library `string`,
I don't think that the (silency) reload of this library is a good idea.

I prefer to protect the lualdap code, like this :
```
	lua_getglobal (L, "string");
	lua_getfield (L, -1, "gsub");
	if (!lua_isfunction (L, -1))
		return faildirect (L, LUALDAP_PREFIX"string.gsub broken");
```

Reply:
I think the protection is not necessary. `lua_call` will fail in that case, hopefully with some useful error message. (Except that we will not know which function we tried to call. But it should tell us that what we tried to call was not a function.)

-> LGTM

Reply:
> If an user remove the standard library `string`,
> I don't think that the (silency) reload of this library is a good idea.

Think of cases where you are not running a Lua interpreter, but where Lua is embedded into another application. The application programmer might not have exposed some (or any) libraries to the Lua scripts.

Reply:
With the protection that I suggest, an embedded will know that `lualdap` depends on `string.gsub`.

The default message tell that you try to call something which is not a function, so it is not easy to find the real problem.